tests/defences/detector/poison/test_spectral_signature_defense.py

```diff
@@ -18,72 +18,61 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import logging
-import unittest
-
+import pytest
 import numpy as np
 
 from art.defences.detector.poison import SpectralSignatureDefense
-from art.utils import load_mnist
 
-from tests.utils import master_seed
+from tests.utils import ARTTestException
 
 logger = logging.getLogger(__name__)
 
-NB_TRAIN, NB_TEST, BATCH_SIZE, EPS_MULTIPLIER, UB_PCT_POISON = 30000, 10, 128, 1.5, 0.2
-
-
-class TestSpectralSignatureDefense(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
 
-        (x_train, y_train), (x_test, y_test), min_, max_ = load_mnist()
-        x_train, y_train = x_train[:NB_TRAIN], y_train[:NB_TRAIN]
-        cls.mnist = (x_train, y_train), (x_test, y_test), (min_, max_)
-        from tests.utils import get_image_classifier_kr
+NB_TRAIN, NB_TEST, BATCH_SIZE, EPS_MULTIPLIER, UB_PCT_POISON = 300, 10, 128, 1.5, 0.2
 
-        cls.classifier = get_image_classifier_kr()
-        cls.defence = SpectralSignatureDefense(
-            cls.classifier,
-            x_train,
-            y_train,
-            batch_size=BATCH_SIZE,
-            eps_multiplier=EPS_MULTIPLIER,
-            ub_pct_poison=UB_PCT_POISON,
-            nb_classes=10,
-        )
 
-    def setUp(self):
-        # Set master seed
-        master_seed(1234)
+@pytest.mark.xfail
```

Comment:
Why is this test expected to fail?

Reply:
The params that are passed in as an argument are created by the `parameterize` method above. All of those params are invalid so this check is to ensure that invalid params are checked by the defense.